core/templates/services/literally-canvas-helper.service.ts

```diff
@@ -58,30 +58,17 @@ angular.module('oppia').factory('LiterallyCanvasHelperService', [
       }
     };
 
-    var validateSVGTag = function(svgString) {
+    var isSvgTagValid = function(svgString) {
       // This function validates an svg tag.
-      var domParser = new DOMParser();
-      var doc = domParser.parseFromString(svgString, 'image/svg+xml');
-      var allowedTags = Object.keys(CONSTANTS.SVG_ATTRS_WHITELIST);
-      var nodeTagName = null;
-      var valid = true;
-      doc.querySelectorAll('*').forEach((node) => {
-        nodeTagName = node.tagName.toLowerCase();
-        if (allowedTags.indexOf(nodeTagName) !== -1) {
-          for (var i = 0; i < node.attributes.length; i++) {
-            if (CONSTANTS.SVG_ATTRS_WHITELIST[nodeTagName].indexOf(
-              node.attributes[i].name.toLowerCase()) === -1) {
-              valid = false;
-            }
-          }
-        } else {
-          valid = false;
-        }
-      });
-      if (!valid) {
+      var dataURI = 'data:image/svg+xml;base64,' + btoa(svgString);
+      var invlidTagsAndAttr = (
+        ImageUploadHelperService.getInvalidSvgTagsAndAttrs(dataURI));
+      if (invlidTagsAndAttr.tags.length === 0 &&
+          invlidTagsAndAttr.attrs.length === 0) {
+        return true;
+      } else {
         throw new Error('Invalid tag or attribute in svg.');
```

Comment:
Does this message need to be made more informative? I'm not sure. Adding @kevintab95 for a decision.

Reply:
Just to give additional information regarding the error. Since the svg's are generated by literallyCanvas the chance of invalid svg being generated is zero unless someone tampers with it. So I believe that the error will never be invoked.

Reply:
OK, thanks. My comment still applies. If we don't expect it to happen in practice, then having more info seems even more important, so that we can debug it if it does happen.

Reply:
Per discussion with @rohitkatlaa, we'll be logging the tags and attrs as well here.

Reply:
Done.